EIPS/eip-XXXX.md

```diff
@@ -0,0 +1,91 @@
+---
+eip: XXX
+title: eth/70 - partial block receipt lists
+description: Adds a facility for paginating block receipts in the p2p protocol
+author: Felix Lange <fjl@ethereum.org>, Jochem Brouwer (@jochem-brouwer), Giulio Rebuffo (@Giulio2002)
+discussions-to:
+status: Final
+type: Standards Track
+category: Networking
+created: 2025-06-16
+requires: 7642, 7825
+---
+
+## Abstract
+
+This EIP modifies the 'eth' p2p protocol to allow requesting partial block receipt lists.
+
+## Motivation
+
+As Ethereum moves toward a higher block gas limit on mainnet, the worst-case total size of
+a block receipts list also becomes larger, and may eventually exceed the 10MB message size
+limit commonly applied in clients. This can lead to sync failures.
+
+## Specification
+
+Modify the encoding for receipts in the `Receipts (0x10)` message as follows:
+
+- (eth/69): `[request-id: P, [[receipt₁, receipt₂], ...]]`
+
+- (eth/70): `[request-id: P, lastBlockIncomplete: {0,1}, [[receipt₁, receipt₂], ...]]`
+
+If the `lastBlockIncomplete` flag is set to true (`1`), the last receipt list does not
+contain all receipts of the block, and the client will have to request the remaining
+receipts of that block in a new request.
+
+To support such partial queries, we also modify the `GetReceipts (0x0f)` message:
+
+- (eth/69): `[request-id: P, [blockhash₁: B_32, blockhash₂: B_32, ...]]`
+
+- (eth/70): `[request-id: P, firstBlockReceiptIndex: P, [blockhash₁: B_32, blockhash₂: B_32, ...]]`
+
+For the first block in the list of requested block hashes, the server should omit receipts
+up to the `firstBlockReceiptIndex` from the response.
+
+Downloading block receipts across multiple messages creates new attack surface. In order
+to defend against malicious servers trying to overload syncing clients, implementations
+should perform additional validation for `Receipts` responses:
+
+- Verify the total number of delivered receipts matches the count of transactions.
+- Verify the size of each receipt against the gas limit of the corresponding transaction,
+  i.e. reject if it is larger than gaslimit/8.
+- Verify the total downloaded receipts size is no larger than allowed by the block gas limit.
+
+## Rationale
+
+<!-- TODO: double check numbers -->
+
+Since [EIP-7825] caps the gas limit of a single transaction to 30M gas, a single
+transaction receipt will always be limited in size. Specifically, a transaction can
+produce at most 30000000/8 = 3.75MB of log data.
+
+However, a block can contain contain multiple transactions, and thus the entire block
+receipts list can be much larger. At a block gas limit of ~80M, the `Receipts` message
```

Comment:
Devp2p has 10 MiB for the practical limit in the spec, so "more correct" would be the gas limit of `10 * 1024 * 1024 * 8`